src/main/java/org/mule/tools/devkit/sonar/checks/java/DevKitAnnotationsOrderCheck.java

```diff
@@ -12,29 +12,31 @@
 
-import com.google.common.collect.Iterables;
+import java.util.List;
+import java.util.regex.Pattern;
 
-@Rule(key = DevKitAnnotationsOrderCheck.KEY, name = "DevKit annotations should be added in a certain order for easing readability", description = "If there is a @Default annotation, it should be the last one (closest to the argument declaration). If there is @RefOnly, it should be prior to @Default.", priority = Priority.CRITICAL, tags = { "connector-certification" })
+@Rule(key = DevKitAnnotationsOrderCheck.KEY, name = "DevKit annotations should be added in a certain order for easing readability", description = "If there is a @Default or @Optional annotation, it should be the last one (closest to the argument declaration). If there is @RefOnly, it should be prior to @Default or @Optional.", priority = Priority.CRITICAL, tags = { "connector-certification" })
 @ActivatedByDefault
 public class DevKitAnnotationsOrderCheck extends AbstractConnectorClassCheck {
 
     public static final String KEY = "devkit-annotations-order";
+    private static final Pattern ORDER_PATTERN = Pattern.compile("^((@RefOnly)*(\\s*)(@Default|@Optional)*$)");
 
     @Override
     protected void verifyProcessor(@NotNull MethodTree tree, @NotNull final IdentifierTree processorAnnotation) {
         for (VariableTree var : tree.parameters()) {
             final List<AnnotationTree> annotations = var.modifiers().annotations();
-            int indexOfDefault = Iterables.indexOf(annotations, ClassParserUtils.hasAnnotationPredicate("org.mule.api.annotations.param.Default"));
-            int indexOfRefOnly = Iterables.indexOf(annotations, ClassParserUtils.hasAnnotationPredicate("org.mule.api.annotations.param.RefOnly"));
-
-            if (indexOfDefault != -1 && indexOfDefault != annotations.size() - 1) {
-                // if there is a @Default annotation it has to be the last one, otherwise raise an issue
-                logAndRaiseIssue(annotations.get(indexOfDefault), String.format("@Default annotation must be the last one in method '%s' argument '%s'.", tree.simpleName(), var.simpleName()));
-            } else if (indexOfRefOnly != -1 && indexOfDefault == -1 && indexOfRefOnly != annotations.size() - 1) {
-                // if there is a @RefOnly annotation it has to be the last one or be right before a @Default one, otherwise raise an issue
-                logAndRaiseIssue(annotations.get(indexOfRefOnly), String.format("@RefOnly annotation must be the last one in method '%s' argument '%s'.", tree.simpleName(), var.simpleName()));
-            } else if (indexOfRefOnly != -1 && indexOfDefault != -1 && indexOfRefOnly != annotations.size() - 2)
-                logAndRaiseIssue(annotations.get(indexOfRefOnly), String.format("@RefOnly annotation must be right before @Default in method '%s' argument '%s'.", tree.simpleName(), var.simpleName()));
+            String unifiedAnnotations = Joiner.on(" ").join(Iterables.transform(annotations,
```

Comment:
Hum... no, this approach won't work, for e.g., if using FQN for annotations, which in the case of Optional can be very reasonable (for example if using Optional from Guava or Optional from Java 8).
On the other hand, it fails to support another annotation at the beginning... for e.g. `@MetaDataKeyParam @RefOnly @Optional`


Please use the former approach of working with indexes, and also add the following tests (and any other if appropriate):
```
    @Processor
    public void aMethodWithRefOnlyAndDefaultFQN(@org.mule.api.annotations.param.RefOnly @org.mule.api.annotations.param.Default Map<String, Object> s1) {
    }

    @Processor
    public void aMethodWithMetaDataKeyParamRefOnlyAndOptional(@MetaDataKeyParam @RefOnly @Optional Map<String, Object> s1) {
    }
```